### DISCUSSION TITLE
balancer status

### DISCUSSION THREAD

**Discussion by StrDa123** [syssi/esphome-pace-bms#18] [Q&A] [answered]
Hello

can you tell me what the number from the balancer status means and how I can decode it?

**Comment by ThHanika** [comment]
Here the answer:
It is a bitset on cell 1-16 (bit0 - bit15) that the passive balancer is passive balancing the cell. If all bit are set at the end of charging, all cells have arrived the ballance-starting voltage. (... and are hopefully complete ballanced  ;-) )

**Comment by jingsno** [comment, marked as answer]
Release 2.0 added support for a "balancer flag" for each cell. When a cell is being (passive) balanced, the value is 'ON'. When charging is finished, and all cells have dropped down to the balancer start voltage, balancing stops, all cell balancer flags / bits are cleared / 'OFF'

**Reply by garylankford8** [reply]
Curious... My balancer reports either 0 or some variable large number... Maybe I need to change something in my yaml?

**Reply by jingsno** [reply]
If you update your ESP code, you'll get a balancer sensor for each battery:
![357727022-1e64219c-1c1e-4e1e-8daa-bfeed228ef21](https://github.com/user-attachments/assets/ccbff329-9005-40bb-99b6-b1ed2e6d142b)

**Reply by jingsno** [reply]
Then you can do something like this, with the lovlace yaml:
![357727396-3b1cc04d-e34f-4542-914f-6e4e2fe35c1c](https://github.com/user-attachments/assets/34081a0f-ab2b-41cc-ae11-c483676bdee8)

**Reply by garylankford8** [reply]
Yeah, I don't get that data from my yaml code. I only get a single balancer number, either 0 or some big number, from each battery...

What is the code I need to add to get the balancing data for each cell.

I'm not smart enough to know or create the code...



Edit: 

OK, I found that apparently new code in the example.yaml.

I really wasn't expecting any more new development on this project. This balancing data was the one thing still sending me to pbmstools, occasionally.

I still would like to see a bit of the code you used in the card to get the colors and conditionals... The whole card YAML would be nice...

I'm still a newbie!

Edit: 

OK, found where these cards came from. Help not needed! Thanks!
